### PR TITLE
Added source/dest check parameter for the ec2_instance type

### PR DESCRIPTION
**Pull request by mlosapio**
https://github.com/puppetlabs/puppetlabs-aws/issues/276

### CODE REVIEW

**Comment by mlosapio** [comment]
I can't figure out vcr and what's wrong here... if someone can help me out I'd appreciate it.

File: lib/puppet/provider/ec2_instance/v2.rb [hunk 1]
```diff
@@ -293,11 +294,28 @@ def create
         )
       end
 
+      ec2.wait_until(:instance_running, instance_ids: instance_ids)
```
Comment (hanikesn) [inline lib/puppet/provider/ec2_instance/v2.rb]: Can't we only wait when we actually want to change the value or do we have to wait in all cases?

**Comment by zleswomp** [comment]
@mlosapio I can help with VCR, but we need to start with a rebase on this one.  Also, more information in the commit would help me understand what the goal here is.  Do you have interest in coming back to this work?

**Comment by mlosapio** [comment]
We can abandon, I dont have the cycles at the moment to come back to this...

**Comment by zleswomp** [comment]
Thanks for coming back with that.  Apologies that this sat for so long @mlosapio.

**Pull request closed without merging**